system-x/services/google/cloud/bigquery/src/main/java/software/tnb/google/cloud/bigquery/validation/BigQueryValidation.java

```diff
@@ -4,97 +4,80 @@
 
 public class BigQueryValidation {
 
     private static final Logger LOG = LoggerFactory.getLogger(BigQueryValidation.class);
-    private BigQuery bigQuery;
-    private String projectId;
+    private final BigQuery bigQuery;
+    private final String projectId;
 
     public BigQueryValidation(BigQuery bigQuery, String projectId) {
         this.bigQuery = bigQuery;
         this.projectId = projectId;
     }
 
-    public void createTableWithSampleSchema(String dataSetId, String tableId) {
-        createTable(dataSetId, tableId, createSampleSchema());
-    }
-
     public void createTable(String dataSetId, String tableId, Schema schema) {
+        LOG.debug("Creating table {}", tableId);
         TableId id = TableId.of(projectId, dataSetId, tableId);
         TableDefinition.Builder builder = StandardTableDefinition.newBuilder().setSchema(schema);
         TableInfo tableInfo = TableInfo.of(id, builder.build());
         bigQuery.create(tableInfo);
     }
 
+    public void createTable(String dataSetId, String tableId, FieldList fieldList) {
+        createTable(dataSetId, tableId, Schema.of(fieldList));
+    }
+
+    public void createTable(String dataSetId, String tableId, Map<String, StandardSQLTypeName> schema) {
+        createTable(dataSetId, tableId, FieldList.of(schema.entrySet().stream()
+            .map(entry -> Field.of(entry.getKey(), entry.getValue())).collect(Collectors.toList())));
+    }
+
     public void deleteTable(String datasetName, String tableName) {
-        LOG.debug("Deleting BQ table " + tableName);
+        LOG.debug("Deleting table {}", tableName);
         bigQuery.delete(TableId.of(projectId, datasetName, tableName));
     }
 
     public void createDataset(String datasetName) {
-        DatasetInfo datasetInfo = DatasetInfo.newBuilder(datasetName).build();
-
-        Dataset newDataset = bigQuery.create(datasetInfo);
-        String newDatasetName = newDataset.getDatasetId().getDataset();
+        LOG.debug("Creating dataset {}", datasetName);
+        bigQuery.create(DatasetInfo.newBuilder(datasetName).build());
     }
 
     public void deleteDataset(String datasetName) {
-        DatasetId datasetId = DatasetId.of(projectId, datasetName);
-        bigQuery.delete(datasetId, BigQuery.DatasetDeleteOption.deleteContents());
+        LOG.debug("Deleting dataset {}", datasetName);
+        bigQuery.delete(DatasetId.of(projectId, datasetName), BigQuery.DatasetDeleteOption.deleteContents());
     }
 
-    public long tableRowsCount(String datasetName, String tableName) {
-        String query =
-            "SELECT * FROM `" + projectId + "." + datasetName + "." + tableName + "`";
-        try {
-            return bigQuery.query(QueryJobConfiguration.of(query)).getTotalRows();
-        } catch (InterruptedException e) {
-            throw new RuntimeException("Unable to query the BQ table", e);
-        }
+    public List<FieldValueList> tableContent(String datasetName, String tableName) {
```

Comment:
If we used big query for *big query* it'd probably be better to return the stream here not to try to load it all into a list... but I doubt we'll delve into big big data :D